galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/routes/RunQueryRoute.java

```diff
@@ -331,39 +333,62 @@ public List<RasRunResult> sortResults(
+		return sortData(runs, queryParams, sortValue);
 		
 	}
 
-	public List<RasRunResult> sortingData(List<RasRunResult> runs, QueryParameters queryParams, @NotNull String sortValue) throws InternalServletException {
-		
-		boolean isToSortAscending = ExtractQuerySort.isAscending(queryParams,"to");
-		boolean isTestClassSortAscending = ExtractQuerySort.isAscending(queryParams,"testclass");
-		boolean isResultSortAscending = ExtractQuerySort.isAscending(queryParams, "result");
-
-		if (sortValue.equals("to:asc") && isToSortAscending) {
-			Collections.reverse(runs);
-		} else if (sortValue.equals("to:desc") &&!isToSortAscending) {
-			return runs;
-		} else if (sortValue.equals("testclass:asc") && isTestClassSortAscending) {
-			Collections.sort(runs, new SortByTestClass());
-		} else if (sortValue.equals("testclass:desc") &&!isTestClassSortAscending) {
-			Collections.sort(runs, new SortByTestClass());
-			Collections.reverse(runs);   
-		} else if (sortValue.equals("result:asc") && isResultSortAscending) {
-			Collections.sort(runs, new SortByResult());
-		} else if (sortValue.equals("result:desc") && !isResultSortAscending) {
-			Collections.sort(runs, new SortByResult());
-			Collections.reverse(runs);
+	public List<RasRunResult> sortData(List<RasRunResult> runs, QueryParameters queryParams, @NotNull String sortValue) throws InternalServletException {
+
+		if (this.SortQueryParameterChecker.validateSortValue(queryParams) || !this.SortQueryParameterChecker.validateSortValue(queryParams)){
+			if (sortValue.toLowerCase().startsWith("to") ) {
+				boolean isAscending = this.SortQueryParameterChecker.isAscending(queryParams,"to");
+				if (isAscending) {
+					Collections.reverse(runs);
+				}
+			} else if (sortValue.toLowerCase().startsWith("testclass")) {
+				boolean isAscending = this.SortQueryParameterChecker.isAscending(queryParams,"testclass");
+				if (isAscending) {
+					Collections.sort(runs, new SortByTestClass());
+				} else {
+					Collections.sort(runs, new SortByTestClass());
+					Collections.reverse(runs);   
+				}
+
+			} else if (sortValue.toLowerCase().startsWith("result")) {
+				boolean isAscending = this.SortQueryParameterChecker.isAscending(queryParams, "result");
+				if (isAscending) {
+					Collections.sort(runs, new SortByResult());
+				} else {
+					Collections.sort(runs, new SortByResult());
+					Collections.reverse(runs);
+				}
+			}
 		}
 		return runs;
 	}
 
 	public String extractSortValue (QueryParameters paramMap) throws InternalServletException {
-		String sortValue = paramMap.getSingleString("sort",null);
-		if (sortValue == null){
-			sortValue = "to:desc";
+		return paramMap.getSingleString("sort","to:desc");
+	}
+
+	public Instant getDefaultFromInstantIfNoQueryIsPresent (QueryParameters paramMap) throws InternalServletException{
+		int querysize = paramMap.getSize();
+		Instant from = null ;
+		if (querysize > 0){
+			from = paramMap.getSingleInstantIfParameterNotPresent("from", "runname");
```

Comment:
Thought we were going to rename that method to assert that one or the other is present. I can't stand that method name.